interoperability-layer-openhim/swarm.sh

```diff
@@ -60,112 +75,119 @@ verify_mongos() {
+function init_package() {
+  (
+    docker::deploy_service "${COMPOSE_FILE_PATH}" "docker-compose-mongo.yml" "$MONGO_CLUSTER_COMPOSE_FILENAME" "$MONGO_DEV_COMPOSE_FILENAME"
+    docker::deploy_sanity "${mongo_services[@]}"
 
-    if [[ "${STATEFUL_NODES}" == "cluster" ]]; then
-      try "${COMPOSE_FILE_PATH}/initiateReplicaSet.sh" "Fatal: Initate Mongo replica set failed."
+    if [[ "${NODE_MODE}" == "cluster" ]]; then
+      try "${COMPOSE_FILE_PATH}/initiate-replica-set.sh" throw "Fatal: Initiate Mongo replica set failed"
     fi
 
     prepare_console_config
 
-    try "docker stack deploy -c ${COMPOSE_FILE_PATH}/docker-compose.yml -c ${COMPOSE_FILE_PATH}/docker-compose.stack-0.yml $openhim_dev_compose_param instant" "Failed to create stack-0 openhim core/console"
+    docker::deploy_service "${COMPOSE_FILE_PATH}" "docker-compose.yml" "docker-compose.stack-0.yml" "$OPENHIM_DEV_COMPOSE_FILENAME"
 
-    try "docker stack deploy -c ${COMPOSE_FILE_PATH}/docker-compose.await-helper.yml instant" "Failed to deploy await-helper"
+    log info "Waiting to give OpenHIM Core time to start up before OpenHIM Console run..."
+    docker::deploy_sanity "openhim-core"
+    config::await_service_running "openhim-core" "${COMPOSE_FILE_PATH}"/docker-compose.await-helper.yml "${OPENHIM_CORE_INSTANCES}"
 
-    log info "Waiting to give OpenHIM Core time to start up before OpenHIM Console run"
-    verify_core
+    docker::deploy_service "${COMPOSE_FILE_PATH}" "docker-compose.yml" "docker-compose.stack-1.yml" "$OPENHIM_DEV_COMPOSE_FILENAME"
+    docker::deploy_sanity "openhim-console"
+  ) ||
+    {
+      log error "Failed to deploy Interoperability Layer OpenHIM package"
+      exit 1
+    }
+
+  docker::deploy_config_importer "$COMPOSE_FILE_PATH/importer/docker-compose.config.yml" "interoperability-layer-openhim-config-importer" "openhim"
+}
 
-    try "docker stack deploy -c ${COMPOSE_FILE_PATH}/docker-compose.yml -c ${COMPOSE_FILE_PATH}/docker-compose.stack-1.yml $openhim_dev_compose_param instant" "Failed to create stack-1 openhim core/console"
+function scale_services_up() {
+  (
+    docker::deploy_service "${COMPOSE_FILE_PATH}" "docker-compose-mongo.yml" "$MONGO_CLUSTER_COMPOSE_FILENAME" "$MONGO_DEV_COMPOSE_FILENAME"
+    docker::deploy_sanity "${mongo_services[@]}"
 
-    try "docker stack deploy -c ${COMPOSE_FILE_PATH}/importer/docker-compose.config.yml instant" "Failed to deploy config importer"
+    verify_mongos
+    prepare_console_config
 
-    log info "Waiting to give core config importer time to run before cleaning up service"
+    docker::deploy_service "${COMPOSE_FILE_PATH}" "docker-compose.yml" "docker-compose.stack-1.yml" "$OPENHIM_DEV_COMPOSE_FILENAME"
+    docker::deploy_sanity "openhim-core" "openhim-console"
+  ) || {
+    log error "Failed to scale up Interoperability Layer OpenHIM package"
+    exit 1
+  }
+}
 
-    config::remove_config_importer interoperability-layer-openhim-config-importer
+function start_package() {
```

Comment:
This entire function can be removed. Put the variable things into the init_vars function, then move the rest into the main function by splitting out the `init` and `up` sections

Reply:
cool done :)